extensions/azure-functions-http/runtime/src/main/java/org/jboss/logging/AzureFunctionsLoggerProvider.java

```diff
@@ -0,0 +1,21 @@
+package org.jboss.logging;
+
+/**
+ * Custom logger provider to plug the logger from Azure functions inside
+ */
+public class AzureFunctionsLoggerProvider extends AbstractMdcLoggerProvider implements LoggerProvider {
+    private static java.util.logging.Logger azureFunctionsLogger;
+
+    public static void setAzureFunctionsLogger(java.util.logging.Logger logger) {
+        azureFunctionsLogger = logger;
```

Comment:
You could put the logger in a ThreadLocal, you just have to make sure to clear the thread local after the request is done.

Reply:
I changed it for a ThreadLocal.

Also:

* added a reset method
* put a try finally block to properly reset ThreadLocal after processing whatever happens


I also added a "backup" process to delegate to a normal Quarkus logger in case there is no azure logger defined, i.e. the execution of the function has spawn a thread in which the logger is no more defined.

But while thinking on that we have a bigger problem. Let's say I want to do a REST client call, that you'd typically use the http client library. The recommendation from Azure in this case is to use a a pool of client connections. But then each of this connections is running on its own thread and client logs wont be sent to the azure function's logstream...

Reply:
depends what client library you are using.  A pool of client connections doesn't mean a pool of threads too.

Reply:
I think the worry of @mederel  is in the case some user code calls async APIs in an Azure function, like a REST client method returning CompletionStage and executing the callback in a separate thread.
I assume the `ThreadLocal<java.util.logging.Logger>` will propagate to the other thread if user follows https://quarkus.io/guides/context-propagation, so any logging in the callback will still be routed to Azure logger ?